data/tabular/train_test_split.py

```diff
@@ -197,40 +214,86 @@ def cli(
     # for path in paths_to_data:
     #     if "flashpoint" in path:
     #         filtered_paths.append(path)
-    #     elif "freesolv" in path:
+    #     elif "BBBP" in path:
     #         filtered_paths.append(path)
     #     elif "peptide" in path:
     #         filtered_paths.append(path)
+    #     elif "bicerano_dataset" in path:
+    #         filtered_paths.append(path)
     # paths_to_data = filtered_paths
 
     REPRESENTATION_LIST = []
 
-    for path in tqdm(paths_to_data):
+    for path in paths_to_data:
         df = pd.read_csv(path)
-        if repr_col in df.columns:
+
+        if repr_col in df.columns and path not in TRACKED_DATASETS:
+            print("Processing", path.split("/")[0])
+            TRACKED_DATASETS.append(path)
             REPRESENTATION_LIST.extend(df[repr_col].to_list())
 
     REPR_DF = pd.DataFrame()
-    REPR_DF["SMILES"] = list(set(REPRESENTATION_LIST))
+    REPR_DF[repr_col] = list(set(REPRESENTATION_LIST))
 
-    scaffold_split = create_scaffold_split(
-        REPR_DF, seed=seed, frac=[train_size, val_size, test_size]
-    )
+    if "SMILES" in REPR_DF.columns:
+        split = create_scaffold_split(
+            REPR_DF, seed=seed, frac=[train_size, val_size, test_size]
+        )
+    else:
+        split_ = pd.DataFrame(
+            np.random.choice(
+                ["train", "test", "valid"],
+                size=len(REPR_DF),
+                p=[1 - val_size - test_size, test_size, val_size],
+            )
+        )
+
+        REPR_DF["split"] = split_
+        train = REPR_DF.query("split == 'train'").reset_index(drop=True)
+        test = REPR_DF.query("split == 'test'").reset_index(drop=True)
+        valid = REPR_DF.query("split == 'valid'").reset_index(drop=True)
+
+        split = {"train": train, "test": test, "valid": valid}
 
     # create train and test dataframes
-    train_df = scaffold_split["train"]
-    test_df = scaffold_split["test"]
+    train_df = split["train"]
+    test_df = split["test"]
+    valid_df = split["valid"]
     # add split columns to train and test dataframes
     train_df["split"] = len(train_df) * ["train"]
     test_df["split"] = len(test_df) * ["test"]
+    valid_df["split"] = len(valid_df) * ["valid"]
 
     # merge train and test across all datasets
-    merge = pd.concat([train_df, test_df], axis=0)
+    merge = pd.concat([train_df, test_df, valid_df], axis=0)
     # rewrite data_clean.csv for each dataset
     rewrite_data_with_splits(
-        paths_to_data, merge, override=override, check=check, repr_col=repr_col
+        csv_paths=paths_to_data,
+        repr_col=repr_col,
+        train_test_df=merge,
+        override=override,
+        check=check,
     )
 
 
+def cli(
```

Comment:
nice, this should do what we want it to do. I'll walk through it once more tomorrow with a fresh mind